src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/DetailControl.xaml.cs

```diff
@@ -67,50 +60,52 @@ public void Refresh()
 
         private void ProjectInstallButtonClicked(object sender, EventArgs e)
         {
-            var model = (DetailControlModel)DataContext;
+            var model = (PackageDetailControlModel)DataContext;
 
-            if (model.SelectedVersion == null)
+            if (model != null && model.SelectedVersion != null)
             {
-                return;
-            }
-
-            var userAction = UserAction.CreateInstallAction(
+                var userAction = UserAction.CreateInstallAction(
                 model.Id,
                 model.SelectedVersion.Version);
-            ExecuteUserAction(userAction, NuGetActionType.Install);
+
+                ExecuteUserAction(userAction, NuGetActionType.Install);
+            }
         }
 
         private void ProjectUninstallButtonClicked(object sender, EventArgs e)
         {
             var model = (PackageDetailControlModel)DataContext;
-            var userAction = UserAction.CreateUnInstallAction(model.Id);
-            ExecuteUserAction(userAction, NuGetActionType.Uninstall);
-        }
 
+            if (model != null)
```

Comment:
this is the problem, since it type cast to Project specific detailmodel where as it's being hit from solution level which is why model is null. Ideally it should be typecast to `DetailControlModel` and then check for `SelectedVersion`

Reply:
I am not sure I follow. The call for solution install is [here](https://github.com/NuGet/NuGet.Client/pull/2253/files/217180da211e72f710f8fcfecdfd55494a23979d#diff-8300d162b02a3c967d463699da1c509dR86) and it is being cast to `PackageSolutionDetailControlModel`.